app/services/training_periods/create.rb

```diff
@@ -0,0 +1,29 @@
+module TrainingPeriods
+  class Create
+    def initialize(period:, started_on:, school_partnership:, expression_of_interest:)
+      @period = period
+      @started_on = started_on
+      @school_partnership = school_partnership
+      @expression_of_interest = expression_of_interest
+    end
+
+    def call
```

Comment:
`#create_training_period` might be a bit more descriptive?

Reply:
I did wonder about that but I went with `.call` mainly to keep it aligned with common service object patterns where the purpose of the class already describes the action (i.e 'Create' is already in the name).
  

`TrainingPeriods::Create.new(...).call` felt a bit more idiomatic and avoids the duplication that `TrainingPeriods::Create.new(...).create_training_period` has.

Happy to go with the longer form if you think clarity would be improved but wanted to flag the reason behind my decision.

Let me know your thoughts.

Reply:
I think the reason is sound but generally prefer the clearer version. I know we're unlikely to do it here in this case but if you had `a = TrainingPeriods::Create.new(...)` then several lines later `a.call`, it would be less clear than `a.create_training_period`